dhis-2/dhis-api/src/main/java/org/hisp/dhis/tracker/imports/validation/ValidationCode.java

```diff
@@ -29,152 +29,141 @@
+  E1301("Generated by ProgramRule (`{0}`) - Mandatory DataElement `{1}` is not present."),
+  E1302("DataElement `{0}` is not valid: `{1}`."),
+  E1303("Mandatory DataElement `{0}` is not present."),
+  E1304("DataElement `{0}` is not a valid data element."),
+  E1305("DataElement `{0}` is not part of ProgramStage `{1}`."),
+  E1306("Generated by ProgramRule (`{0}`) - Mandatory Attribute `{1}` is not present"),
   E1307(
-      "Generated by program rule (`{0}`) - Unable to assign value to data element `{1}`. "
+      "Generated by ProgramRule (`{0}`) - Unable to assign value to DataElement `{1}`. "
           + "The provided value must be empty or match the calculated value `{2}`"),
-  E1308("Generated by program rule (`{0}`) - DataElement `{1}` is being replaced in event `{2}`"),
+  E1308("Generated by ProgramRule (`{0}`) - DataElement `{1}` is replaced in event `{2}`."),
   E1309(
-      "Generated by program rule (`{0}`) - Unable to assign value to attribute `{1}`. "
-          + "The provided value must be empty or match the calculated value `{2}`"),
-  E1310("Generated by program rule (`{0}`) - Attribute `{1}` is being replaced in te `{2}`"),
-  E1313(
-      "Event {0} of an enrollment does not point to an existing tracked entity. The data in your system might be corrupted"),
-  E1314("Generated by program rule (`{0}`) - DataElement `{1}` is mandatory and cannot be deleted"),
-  E1315(
-      "Status `{0}` does not allow defining data values. Statuses that do allow defining data values are: {1}"),
-  E1316("No event can transition from status `{0}` to status `{1}`."),
-  E1317("Generated by program rule (`{0}`) - Attribute `{1}` is mandatory and cannot be deleted"),
+      "Generated by ProgramRule (`{0}`) - Unable to assign value to Attribute `{1}`. "
+          + "The provided value must be empty or match the calculated value `{2}`."),
+  E1310("Generated by ProgramRule (`{0}`) - Attribute `{1}` is being replaced in te `{2}`."),
+  E1313("Event `{0}` of an Enrollment does not reference a TrackedEntity."),
+  E1314("Generated by ProgramRule (`{0}`) - DataElement `{1}` is mandatory and cannot be deleted."),
+  E1315("Status `{0}` does not allow creating data values, allow statuses: {1}."),
+  E1316("Event cannot transition from status `{0}` to status `{1}`."),
+  E1317("Generated by ProgramRule (`{0}`) - Attribute `{1}` is mandatory and cannot be deleted."),
 
   /* Relationship */
-  E4000("Relationship: `{0}` cannot link to itself"),
+  E4000("Relationship: `{0}` cannot link to itself."),
   E4001(
-      "Relationship item `{0}` for relationship `{1}` is invalid: an item must link exactly one of trackedEntity, enrollment, event."),
-  E4006("Could not find relationship type: `{0}`."),
-  E4010("Relationship type `{0}` constraint requires a {1} but a {2} was found."),
+      "RelationshipItem `{0}` for Relationship `{1}` is invalid as an item must be associated with exactly one of TrackedEntity, Enrollment, Event."),
```

Comment:
In other places in this file `TrackedEntity`, `Enrollment` and `Event` start with a lower case.

Reply:
Agreed, fixed several occurrences now. There is one code where it refers to a property, not the entity, so I left it starting with lower-case.